src/set_theory/game/nim.lean

```diff
@@ -265,12 +265,26 @@ by rw [←grundy_value_eq_iff_equiv, grundy_value_zero]
 
 lemma grundy_value_star : grundy_value star = 1 := by simp [nim.nim_one_equiv.symm]
 
+@[simp] lemma grundy_value_neg (G : pgame) [G.impartial] : grundy_value (-G) = grundy_value G :=
+by rw [grundy_value_eq_iff_equiv_nim, neg_equiv_iff, nim.neg_nim, ←grundy_value_eq_iff_equiv_nim]
+
+lemma grundy_value_eq_mex_right : ∀ (G : pgame) [G.impartial],
+  grundy_value G = ordinal.mex.{u u} (λ i, grundy_value (G.move_right i))
+| ⟨l, r, L, R⟩ := begin
+  introI H,
+  rw [←grundy_value_neg, grundy_value_eq_mex_left],
+  congr,
+  ext i,
+  haveI : (R i).impartial := @impartial.move_right_impartial ⟨l, r, L, R⟩ _ i,
+  apply grundy_value_neg
+end
```

Comment:
This proof feels very hacky, but after some experimenting I couldn't find a better way. I wonder whether there is some missing API that would make this proof less painful.

Reply:
The usual proof of this fact, which would be "right moves are the same as left moves and done", doesn't work here since we went for a more general definition of impartial games than that in the literature (it's usually required that `x = -x`, we just require every left move is equivalent to some right move and viceversa).